packages/apollo/src/updaters/removeFromList.ts

```diff
@@ -26,10 +24,19 @@ export default function removeFromList<Result, Vars = {}>(
       }
     }
 
-    set(
-      nextResult,
+    const rootItem = listPath.split('.')[0];
+    // @ts-ignore
+    const resultRoot = queryResult[rootItem] as StoreObject;
+    // Evict cache before writeQuery: https://github.com/apollographql/apollo-client/issues/6451
+    cache.evict({
+      id: cache.identify(resultRoot),
+      broadcast: false,
+    });
```

Comment:
As stated here https://github.com/apollographql/apollo-client/issues/6451#issuecomment-645612063 we need to evict cache when removing items from a list